discovery_climate_music_analysis/notebooks/2_chart_analysis.py

```diff
@@ -0,0 +1,198 @@
+# ---
+# jupyter:
+#   jupytext:
+#     cell_metadata_filter: -all
+#     comment_magics: true
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.14.5
+#   kernelspec:
+#     display_name: discovery_climate_music_analysis
+#     language: python
+#     name: discovery_climate_music_analysis
+# ---
+
+# %% [markdown]
+# # Chart Analysis
+
+# %%
+import numpy as np
+import pandas as pd
+import altair as alt
+import configparser
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from pprint import pprint
+from dateutil.parser import parse
+
+from discovery_climate_music_analysis.getters.collect_songs import create_dataframe
+from discovery_climate_music_analysis.getters.collect_chart_songs import load_s3_data
+from jacc_hammer.fuzzy_hash import Cos_config, Fuzzy_config, match_names_stream
+
+USERNAME = "addie234"
+PLAYLIST_ID = "spotify:playlist:4EhBYkl4jtI2POrkUxU6Ul"
+
+
+bucket_name = "discovery-hub-open-data"
+
+# %%
+# read chart data
+all_top_charts = load_s3_data(
+    bucket_name, "climate_change_environment_songs/chart_songs.csv"
+)
+
+# %%
+all_top_charts = all_top_charts.columns.to_frame().T.append(
+    all_top_charts, ignore_index=True
+)
+all_top_charts.columns = ["week_start", "chart_pos", "artist", "song"]
+all_top_charts.reset_index(drop=True, inplace=True)
+
+# %% [markdown]
+# ## Have any of our songs charted?
+
+# %%
+# set up for fuzzy matching
+threshold = 0.3
+chunksize = 100_000
+
+cos_config = Cos_config()
+fuzzy_config = Fuzzy_config()
+
+tmp_dir = Path(TemporaryDirectory().name)
+tmp_dir.mkdir()
+
+# %%
+# song plus artist for both lists
+
+# x
+list_of_chart_songs = (all_top_charts["song"] + " " + all_top_charts["artist"]).values
+# y
+list_of_spotify_songs = (spotify_df_["name"] + " " + spotify_df_["artist"]).values
```

Comment:
Hey, I think you haven't defined `spotify_df_` yet in this notebook. Probably missing a line with something like:
```
spotify_df_ = create_dataframe(USERNAME, PLAYLIST_ID)
```